lib/proxy/validations.rb

```diff
@@ -14,8 +16,17 @@ def valid_mac? mac
 
   # validates the ip address
   def validate_ip ip
-    raise Error, "Invalid IP Address #{ip}" unless ip =~ /(\d{1,3}\.){3}\d{1,3}/
+    IPAddr.new(ip)
     ip
+  rescue IPAddr::InvalidAddressError
```

Comment:
The exception thrown differs between ruby versions, please add something like this: https://github.com/theforeman/foreman/blob/79c0664a008afe0fea54a0b5256426f3867a02bc/app/models/subnet.rb#L288-L292